spec/text-editor-spec.coffee

```diff
@@ -1986,6 +1986,82 @@ describe "TextEditor", ->
         expect(editor2.getSelectedBufferRanges()).not.toEqual editor.getSelectedBufferRanges()
 
   describe "buffer manipulation", ->
+    describe ".moveLineUp", ->
+      describe "when there is only one line selected", ->
+        it "moves the line up by one row", ->
+          editor.setSelectedBufferRange([[3, 2], [3, 9]])
+          expect(editor.getSelectedBufferRange()).toEqual {start: {row: 3, column: 2}, end: {row: 3, column: 9}}
```

Comment:
Wow, that's odd. There may be some method that's not doing a `Range.fromObject` on its arguments and causing issues, but the equality checks should work. Could you try one more time with this particular assertion and if it doesn't work I'll investigate?